src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/permissions/PermissionsClient.scala

```diff
@@ -1,98 +1,86 @@
 import org.broadinstitute.dsde.agora.server.model.AgoraEntity
+import slick.jdbc.JdbcProfile
 
-import scala.concurrent.{Await, Future}
 import scala.concurrent.ExecutionContext.Implicits.global
 import scala.concurrent.duration._
-import scala.util.Try
+import scala.util.{Failure, Success, Try}
 
-trait PermissionsClient {
-  import AgoraConfig.sqlDatabase.profile.api._
-  val db = AgoraConfig.sqlDatabase.db
+abstract class PermissionsClient(profile: JdbcProfile) {
+  import profile.api._
 
   val timeout = 10.seconds
 
   def alias(entity: AgoraEntity): String
 
+  def withPermissionNotFoundException[T](errorString: String = "Could not get permission")( op: => ReadWriteAction[T] ): ReadWriteAction[T] = {
+    op.asTry flatMap {
+      case Success(permission) => DBIO.successful(permission)
+      case Failure(ex) => DBIO.failed(new PermissionNotFoundException(errorString, ex))
+    }
+  }
+
   // Users
-  def addUserIfNotInDatabase(userEmail: String): Unit = {
+  def addUserIfNotInDatabase(userEmail: String): WriteAction[Int] = {
     // Attempts to add user to UserTable and ignores errors if user already exists
-    try {
-      Await.ready(db.run(users += UserDao(userEmail)), timeout)
-    } catch {
-      case _ : Throwable => //Do nothing
+    (users += UserDao(userEmail)).asTry flatMap {
+      case Success(a) => DBIO.successful(a)
+      case Failure(regret) => DBIO.successful(0)
     }
   }
 
-  def isAdmin(userEmail: String): Boolean = {
-    val userQuery = users.findByEmail(userEmail)
-    val user = Await.result(db.run(userQuery.result.head), timeout)
-    user.isAdmin
+  def isAdmin(userEmail: String): ReadAction[Boolean] = {
+    users.findByEmail(userEmail).result map { aa =>
+      aa.head.isAdmin
+    }
   }
 
-  def listAdmins(): Seq[String] = {
-    val adminsQuery = for {
-      user <- users if user.is_admin === true
-    } yield user.email
+  def listAdmins(): ReadWriteAction[Seq[String]] = {
+    withPermissionNotFoundException() {
+      val adminsQuery = for {
+        user <- users if user.is_admin === true
+      } yield user.email
 
-    try {
-      Await.result(db.run(adminsQuery.result), timeout)
-    } catch {
-      case ex: Throwable =>
-        throw new PermissionNotFoundException(ex.getMessage, ex)
+      adminsQuery.result
     }
   }
 
-  def updateAdmin(userEmail: String, adminStatus: Boolean) = {
-    addUserIfNotInDatabase(userEmail)
-
-    // construct update action
-    val adminsUpdateAction = for {
-      user <- users
-        .filter(_.email === userEmail)
-        .map(_.is_admin)
-        .update(adminStatus)
-    } yield user
-
-    // run update action
-    try {
-      val rowsEdited = Await.result(db.run(adminsUpdateAction), timeout)
-
-      if (rowsEdited == 0)
-        throw new Exception("No rows were edited.")
-      else
-        rowsEdited
-
-    } catch {
-      case ex: Throwable => throw new PermissionNotFoundException(s"Could not make user ${userEmail} admin", ex)
```

Comment:
isn't it nice that all this try/catch nonsense has moved into `withPermissionNotFoundException`? :)